### ISSUE TITLE
No auto-redirect if token expired

### ISSUE THREAD

**Issue by alexmt** [akuity/kargo#2237]
# Checklist

* [x] I've searched the issue queue to verify this is not a duplicate bug report.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `kargo version`.
* [x] I've pasted logs, if applicable.

# Description

# Screenshots

<img width="1679" alt="image" src="https://github.com/akuity/kargo/assets/426437/7bcf6b35-faee-4722-85c1-5db0487ef007">

# Steps to Reproduce

1. Configure SSO and login to Kargo
2. Wait until token is expired
3. Open Kargo again
4. Error message + loading screen is shown instead of redirecting to login page

# Version

```
v0.7.0
```

**Comment by rbreeze** [comment]
https://github.com/akuity/kargo/pull/2282 Did not in fact address this and introduced an infinite redirect loop. This change will be reverted in #2374.

Revisiting this issue, I think it's possible it's a backend problem, as I just noticed the `error_description` in the screenshot: 

```
Refresh token is invalid or has already been claimed by another client.
```

**Comment by rbreeze** [comment]
Misunderstood what was happening and did some further testing. Added a commit to this PR which will fix this problem: #2374
